Source/Scene/KTX2Transcoder.js

```diff
@@ -0,0 +1,90 @@
+import CompressedTextureBuffer from "../Core/CompressedTextureBuffer.js";
+import defined from "../Core/defined.js";
+import TaskProcessor from "../Core/TaskProcessor.js";
+
+/**
+ * Transcodes KTX2 textures using web workers.
+ *
+ * @private
+ */
+function KTX2Transcoder() {}
+
+KTX2Transcoder._transcodeTaskProcessor = new TaskProcessor(
+  "transcodeKTX2",
+  Number.POSITIVE_INFINITY // KTX2 transcoding is used in place of Resource.fetchImage, so it can't reject as "just soooo busy right now"
+);
+
+KTX2Transcoder._readyPromise = undefined;
+
+function makeReadyPromise() {
+  var readyPromise = KTX2Transcoder._transcodeTaskProcessor
+    .initWebAssemblyModule({
+      modulePath: "ThirdParty/Workers/basis_transcoder.js",
+      wasmBinaryFile: "ThirdParty/basis_transcoder.wasm",
+    })
+    .then(function () {
+      return KTX2Transcoder._transcodeTaskProcessor;
+    });
+  KTX2Transcoder._readyPromise = readyPromise;
+}
+
+KTX2Transcoder.transcode = function (ktx2Buffer, supportedTargetFormats) {
+  if (!defined(KTX2Transcoder._readyPromise)) {
+    makeReadyPromise();
+  }
+
+  return KTX2Transcoder._readyPromise
+    .then(function (taskProcessor) {
+      var parameters;
+      if (ktx2Buffer instanceof ArrayBuffer) {
+        var view = new Uint8Array(ktx2Buffer);
+        parameters = {
+          supportedTargetFormats: supportedTargetFormats,
+          ktx2Buffer: view,
+        };
+        return taskProcessor.scheduleTask(parameters, [ktx2Buffer]);
+      }
+      parameters = {
+        supportedTargetFormats: supportedTargetFormats,
+        ktx2Buffer: ktx2Buffer,
+      };
+      return taskProcessor.scheduleTask(parameters, [ktx2Buffer.buffer]);
+    })
+    .then(function (result) {
+      var levelsLength = result.length;
+      var faceKeys = Object.keys(result[0]);
```

Comment:
Is `Object.keys` required? Is `result` an array?

Reply:
`result` is an array of objects, so this returns an array of the `positiveX ... negativeZ` face keys.